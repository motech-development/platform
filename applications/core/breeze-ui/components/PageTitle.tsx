import { FC } from 'react';
import Typography from './Typography';

type TColour = 'primary' | 'secondary';

const styles = (colour: TColour) => {
  let colourStyles: string;

  if (colour === 'primary') {
    colourStyles = 'bg-gray-900';
  } else {
    colourStyles = 'bg-gray-200';
  }

  return colourStyles;
};

const PageTitle: FC<{
  colour?: TColour;
  subTitle?: string;
  title: string;
}> = ({ colour = 'primary', subTitle, title }) => (
  <header className={`pb-36 -mb-36 ${styles(colour)}`}>
    <div className="max-w-7xl mx-auto p-4 sm:p-6 lg:p-8">
      <Typography component="h2" variant="h2" margin="sm">
        <span className="decoration-clone bg-blue-600 text-gray-100 p-1">
          {title}
        </span>
      </Typography>

      {subTitle && (
        <Typography component="p" variant="lead" margin="none">
          <span className="decoration-clone bg-gray-100 text-gray-900 p-1">
            {subTitle}
          </span>
        </Typography>
      )}
    </div>
  </header>
);

export default PageTitle;
