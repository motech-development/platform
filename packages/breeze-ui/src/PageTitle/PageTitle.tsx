import { FC, memo } from 'react';
import styled from 'styled-components';
import Typography from '../Typography/Typography';

const PageTitleTypography = styled(Typography)`
  line-height: 1.4;
  margin: 0 0 0.2rem;
`;

const TitleInner = styled.span`
  background-color: #007fa8;
  box-decoration-break: clone;
  color: #fff;
  display: inline;
  padding: 0.25rem 0.5rem;
`;

const SubTitleInner = styled.span`
  background-color: #f8f8f8;
  box-decoration-break: clone;
  color: #000;
  display: inline;
  padding: 0.25rem 0.5rem;
`;

const Split = styled.hr`
  border: 0;
  margin: 0;
  padding: 0;
`;

const Wrapper = styled.div`
  margin-bottom: 1rem;
  padding: 0 0.25rem 0 0;
`;

export interface IPageTitleProps {
  subTitle?: string;
  title: string;
}

const PageTitle: FC<IPageTitleProps> = ({ subTitle = null, title }) => (
  <Wrapper>
    <PageTitleTypography component="h2" variant="h2">
      <TitleInner>{title}</TitleInner>
    </PageTitleTypography>

    <Split />

    {subTitle && (
      <PageTitleTypography component="p" variant="lead" margin="none">
        <SubTitleInner>{subTitle}</SubTitleInner>
      </PageTitleTypography>
    )}
  </Wrapper>
);

export default memo(PageTitle);
