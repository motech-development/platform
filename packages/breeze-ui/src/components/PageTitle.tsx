import { FC } from 'react';
import { classNames, themeClass } from '../utils/className';
import TTheme from '../utils/theme';
import Typography from './Typography';

export interface IPageTitleProps {
  colour?: TTheme;
  subTitle?: string;
  title: string;
}

const PageTitle: FC<IPageTitleProps> = ({
  colour = 'primary',
  subTitle,
  title,
}) => (
  <>
    <Typography component="h2" variant="h2" margin="sm">
      {/* @tailwind: bg-blue-600 bg-gray-600 bg-green-600 bg-red-600 bg-yellow-600 */}
      <span
        className={classNames(
          'decoration-clone text-gray-100 px-2 py-1',
          themeClass(colour, 'bg-{theme}-600'),
        )}
      >
        {title}
      </span>
    </Typography>

    {subTitle && (
      <Typography component="p" variant="lead" margin="none">
        <span className="decoration-clone bg-gray-100 text-gray-900 px-2 py-1">
          {subTitle}
        </span>
      </Typography>
    )}
  </>
);

export default PageTitle;
