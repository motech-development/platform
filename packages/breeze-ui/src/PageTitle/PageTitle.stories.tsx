import { text, withKnobs } from '@storybook/addon-knobs';
import BaseStyles from '../BaseStyles/BaseStyles';
import PageTitle from './PageTitle';

export default {
  component: PageTitle,
  decorators: [withKnobs],
};

export const BasicTypography = {
  name: 'Basic typography',
  render: () => (
    <>
      <BaseStyles />

      <PageTitle
        title={text('Title', 'Page title')}
        subTitle={text('Subtitle', '')}
      />
    </>
  ),
};
