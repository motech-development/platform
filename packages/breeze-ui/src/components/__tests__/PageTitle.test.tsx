import { render } from '@testing-library/react';
import PageTitle from '../PageTitle';

describe('PageTitle', () => {
  it('should display only a title', () => {
    const { container } = render(<PageTitle title="Page title" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render title and a sub title', () => {
    const { container } = render(
      <PageTitle title="Page title" subTitle="Sub title" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('theme', () => {
    it('should render primary theme styles', () => {
      const { container } = render(
        <PageTitle colour="primary" title="Page title" subTitle="Sub title" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render secondary theme styles', () => {
      const { container } = render(
        <PageTitle
          colour="secondary"
          title="Page title"
          subTitle="Sub title"
        />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render success theme styles', () => {
      const { container } = render(
        <PageTitle colour="success" title="Page title" subTitle="Sub title" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render warning theme styles', () => {
      const { container } = render(
        <PageTitle colour="warning" title="Page title" subTitle="Sub title" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render danger theme styles', () => {
      const { container } = render(
        <PageTitle colour="danger" title="Page title" subTitle="Sub title" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
