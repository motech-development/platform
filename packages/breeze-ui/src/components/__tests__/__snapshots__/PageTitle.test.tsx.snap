// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should display only a title 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-blue-600"
    >
      Page title
    </span>
  </h2>
</div>
`;

exports[`PageTitle should render title and a sub title 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-blue-600"
    >
      Page title
    </span>
  </h2>
</div>
`;

exports[`PageTitle theme should render danger theme styles 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-red-600"
    >
      Page title
    </span>
  </h2>
</div>
`;

exports[`PageTitle theme should render primary theme styles 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-blue-600"
    >
      Page title
    </span>
  </h2>
</div>
`;

exports[`PageTitle theme should render secondary theme styles 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-gray-600"
    >
      Page title
    </span>
  </h2>
</div>
`;

exports[`PageTitle theme should render success theme styles 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-green-600"
    >
      Page title
    </span>
  </h2>
</div>
`;

exports[`PageTitle theme should render warning theme styles 1`] = `
<div>
  <h2
    class="mb-2 text-left font-display text-4xl font-semibold"
  >
    <span
      class="decoration-clone text-gray-100 px-2 py-1 bg-yellow-600"
    >
      Page title
    </span>
  </h2>
</div>
`;
